native/src/main/java/io/ballerina/lib/data/yaml/emitter/Emitter.java

```diff
@@ -0,0 +1,317 @@
+                line.append(writeNode(state, scalarEvent.getValue(), event.getTag()));
+            }
+
+            if (event.getKind() == YamlEvent.EventKind.START_EVENT) {
+                YamlEvent.StartEvent startEvent = (YamlEvent.StartEvent) event;
+                if (startEvent.getStartType() == Types.Collection.SEQUENCE) {
+                    line.append(writeFlowSequence(state, event.getTag()));
+                } else if (startEvent.getStartType() == Types.Collection.MAPPING) {
+                    line.append(writeFlowMapping(state, event.getTag()));
+                }
+            }
+
+            event = getEvent(state);
+            isFirstValue = false;
+        }
+
+        line.append("}");
+        return line.toString();
+    }
+
+    private static void writeBlockSequence(EmitterState state, String whitespace, String tag) {
+        YamlEvent event = getEvent(state);
+        boolean emptySequence = true;
+
+        while (true) {
+            if (event.getKind() == YamlEvent.EventKind.END_EVENT) {
+                if (emptySequence) {
+                    state.addLine(whitespace + writeNode(state, "-", tag, true));
+                }
+                break;
+            }
+
+            if (event.getKind() == YamlEvent.EventKind.SCALAR_EVENT) {
+                YamlEvent.ScalarEvent scalarEvent = (YamlEvent.ScalarEvent) event;
+                state.addLine(whitespace + "- " + writeNode(state, scalarEvent.getValue(), event.getTag()));
+            }
+
+            if (event.getKind() == YamlEvent.EventKind.START_EVENT) {
+                YamlEvent.StartEvent startEvent = (YamlEvent.StartEvent) event;
+                if (startEvent.getStartType() == Types.Collection.SEQUENCE) {
+                    if (startEvent.isFlowStyle()) {
+                        state.addLine(whitespace + "- " + writeFlowSequence(state, event.getTag()));
+                    } else {
+                        state.addLine(whitespace + writeNode(state, "-", event.getTag(), true));
+                        writeBlockSequence(state, whitespace + state.indent, event.getTag());
+                    }
+                } else if (startEvent.getStartType() == Types.Collection.MAPPING) {
+                    if (startEvent.isFlowStyle()) {
+                        state.addLine(whitespace + "- " + writeFlowMapping(state, event.getTag()));
+                    } else {
+                        state.addLine(whitespace + "-");
+                        writeBlockMapping(state, whitespace + state.indent);
+                    }
+                }
+            }
+
+            event = getEvent(state);
+            emptySequence = false;
+        }
+    }
+
+    private static String writeFlowSequence(EmitterState state, String tag) {
+        StringBuilder line = new StringBuilder(writeNode(state, "[", tag));
+        YamlEvent event = getEvent(state);
+        boolean firstValue = true;
+
+        while (true) {
```

Comment:
```suggestion
        while (event.getKind() != YamlEvent.EventKind.END_EVENT) {
```

Reply:
No we cannot do this because we have to assign line to empty.